src/test/java/bio/terra/workspace/service/iam/SamServiceTest.java

```diff
@@ -231,6 +242,97 @@ void ListWorkspacesIncludesWsmWorkspace() {
     assertTrue(samWorkspaceIdList.contains(workspaceId));
   }
 
+  @Test
```

Comment:
Do we need tests for the auto-registration code or is it just exercised all of the time?

Reply:
It gets exercised every time WSM starts, added a test to verify registration status